### PR TITLE
elgg_log with level WARNING

### PR DESCRIPTION
**Pull request by raghukul01**
elgg_log with level WARNING if metadata value length is more than x characters

### CODE REVIEW

**Comment by raghukul01** [comment]
#10887

**Comment by raghukul01** [comment]
screenshot 9
what kind of error is this and how can I resolve it?

**Comment by beck24** [comment]
mysql text field has a limit of 65,535 characters

You definitely don't want to log every time something exceeds 50 characters, you'll be filling up the log real fast.  Need a sensible error message so that someone reading the log will know what the issue is.

The travis issue is because of your commit message doesn't meet the expected format: http://learn.elgg.org/en/stable/contribute/code.html#commit-message-format

**Comment by mrclay** [comment]
Is it 65,535 chars or bytes? If the former, we'd need elgg_strlen() instead.

This code should be in Elgg\Database\MetadataTable in create() and update(), right before the queries. The warning message should include the metadata name.

**Comment by mrclay** [comment]
A good commit message would be:

```
fix(metadata): warn devs when saving data longer than column holds

Fixes #10861
```

File: engine/lib/metadata.php [hunk 1]
```diff
@@ -91,6 +91,11 @@ function create_metadata($entity_guid, $name, $value, $value_type = '', $owner_g
  * @return bool
  */
 function update_metadata($id, $name, $value, $value_type, $owner_guid, $access_id) {
+	
+	if (strlen($value) >= 50) {
```
Comment (mrclay) [inline engine/lib/metadata.php]: Replace 50 with 65535.

File: engine/lib/metadata.php [hunk 2]
```diff
@@ -91,6 +91,11 @@ function create_metadata($entity_guid, $name, $value, $value_type = '', $owner_g
...
+		elgg_log(" ", 'WARNING');
```
Comment (mrclay) [inline engine/lib/metadata.php]: Replace `" "` with `"Metadata '{$name}' may be truncated by MySQL's TEXT column."`

File: engine/lib/metadata.php (hunk 1 above)
Comment (mrclay) [inline engine/lib/metadata.php]: And `>=` with `>`

**Comment by mrclay** [comment]
@raghukul01 If you abandon a pull request, please close it.

**Comment by raghukul01** [comment]
ok

**Pull request closed without merging**